go-controller/pkg/ovn/default_network_controller.go

```diff
@@ -741,22 +779,27 @@ func (h *defaultNetworkControllerEventHandler) AddResource(obj interface{}, from
 
 	case factory.EgressNodeType:
 		node := obj.(*kapi.Node)
+		// Add the node to the tracker only if it belongs to the local zone.
+		klog.Infof("SURYA %v", config.OVNKubernetesFeature.EnableInterconnect)
+		if config.OVNKubernetesFeature.EnableInterconnect {
+			// handle zone annotation update on the node
+			if h.oc.isLocalZoneNode(node) {
+				klog.Infof("SURYA %v", node.Name)
+				h.oc.eIPC.localZoneNodes.LoadOrStore(node.Name, true)
+			} else {
+				klog.Infof("SURYA %v", node.Name)
+				h.oc.eIPC.localZoneNodes.Delete(node.Name)
+			}
+		}
 		if err := h.oc.setupNodeForEgress(node); err != nil {
```

Comment:
we need to process nodes only if they are local to this zone, no need to care about other nodes.